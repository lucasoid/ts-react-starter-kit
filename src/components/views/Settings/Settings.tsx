import * as React from 'react';
import { ThemeContext, Themes } from '~theme';
import { View } from '~components/Views/View';
import { Button } from '~components/ui/Button';

export const Settings: React.FC<{}> = () => (
    <ThemeContext.Consumer>
        {({ theme, selectTheme }) => (
            <View>
                <h2>Settings</h2>
                <h3>Theme</h3>
                {Object.keys(Themes).map(key => (
                    <p key={key}>
                        <Button
                            plaintext={true}
                            onClick={() => selectTheme(Themes[key])}
                            styles={{ marginRight: '1em' }}
                        >
                            {Themes[key] === theme ? '☑' : '☐'}
                        </Button>
                        {Themes[key]}
                    </p>
                ))}
            </View>
        )}
    </ThemeContext.Consumer>
);
