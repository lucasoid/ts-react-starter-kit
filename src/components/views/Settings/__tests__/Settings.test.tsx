import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { Settings } from '../Settings';
import { Theme } from '~theme';

describe('Settings', () => {
    it('displays a settings page', () => {
        let wrapper = shallow(
            <Theme>
                <Settings />
            </Theme>,
        ).dive();
        expect(wrapper).toMatchSnapshot();
    });

    it('allows for selecting a theme', () => {
        let wrapper = mount(
            <Theme>
                <Settings />
            </Theme>,
        );
        wrapper
            .find('button')
            .first()
            .simulate('click');
        expect(wrapper).toMatchSnapshot();
        wrapper
            .find('button')
            .at(1)
            .simulate('click');
        expect(wrapper).toMatchSnapshot();
    });
});
