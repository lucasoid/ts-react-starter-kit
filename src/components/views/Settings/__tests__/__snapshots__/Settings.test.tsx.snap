// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings allows for selecting a theme 1`] = `
<Theme>
  <Settings>
    <View>
      <div
        className="view dark"
      >
        <ErrorBoundary>
          <h2>
            Settings
          </h2>
          <h3>
            Theme
          </h3>
          <p
            key="DARK"
          >
            <Button
              onClick={[Function]}
              plaintext={true}
              styles={
                Object {
                  "marginRight": "1em",
                }
              }
            >
              <button
                className="btnPlaintext secondary"
                onClick={[Function]}
                style={
                  Object {
                    "marginRight": "1em",
                  }
                }
              >
                ☑
              </button>
            </Button>
            dark
          </p>
          <p
            key="LIGHT"
          >
            <Button
              onClick={[Function]}
              plaintext={true}
              styles={
                Object {
                  "marginRight": "1em",
                }
              }
            >
              <button
                className="btnPlaintext secondary"
                onClick={[Function]}
                style={
                  Object {
                    "marginRight": "1em",
                  }
                }
              >
                ☐
              </button>
            </Button>
            light
          </p>
        </ErrorBoundary>
      </div>
    </View>
  </Settings>
</Theme>
`;

exports[`Settings allows for selecting a theme 2`] = `
<Theme>
  <Settings>
    <View>
      <div
        className="view light"
      >
        <ErrorBoundary>
          <h2>
            Settings
          </h2>
          <h3>
            Theme
          </h3>
          <p
            key="DARK"
          >
            <Button
              onClick={[Function]}
              plaintext={true}
              styles={
                Object {
                  "marginRight": "1em",
                }
              }
            >
              <button
                className="btnPlaintext secondary"
                onClick={[Function]}
                style={
                  Object {
                    "marginRight": "1em",
                  }
                }
              >
                ☐
              </button>
            </Button>
            dark
          </p>
          <p
            key="LIGHT"
          >
            <Button
              onClick={[Function]}
              plaintext={true}
              styles={
                Object {
                  "marginRight": "1em",
                }
              }
            >
              <button
                className="btnPlaintext secondary"
                onClick={[Function]}
                style={
                  Object {
                    "marginRight": "1em",
                  }
                }
              >
                ☑
              </button>
            </Button>
            light
          </p>
        </ErrorBoundary>
      </div>
    </View>
  </Settings>
</Theme>
`;

exports[`Settings displays a settings page 1`] = `<Settings />`;
